.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,64 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
```

Comment:
I think it might make sense to run this check on any PR, irrespective of the target branch (these are github workflows we do not current use, but still)

Reply:
Not sure. Let's wait for additional input?

Reply:
#42